lib/include/modelSpec.h

```diff
@@ -460,89 +457,57 @@ class NNmodel
     /*! \tparam CurrentSourceModel type of current source model (derived from CurrentSourceModels::Base).
         \param name string containing unique name of current source.
         \param model current source model to use for current source.
+        \param targetNeuronGroupName string name of the target neuron group
         \param paramValues parameters for model wrapped in CurrentSourceModel::ParamValues object.
         \param varInitialisers state variable initialiser snippets and parameters wrapped in CurrentSource::VarValues object.
         \return pointer to newly created CurrentSource */
     template<typename CurrentSourceModel>
-    CurrentSource *addCurrentSource(const string &name, const CurrentSourceModel *model,
+    CurrentSource *addCurrentSource(const string &currentSourceName, const CurrentSourceModel *model,
+                                     const string &targetNeuronGroupName,
                                      const typename CurrentSourceModel::ParamValues &paramValues,
-                                     const typename CurrentSourceModel::VarValues &varInitialisers,
-                                     int hostID = 0, int deviceID = 0)
+                                     const typename CurrentSourceModel::VarValues &varInitialisers)
     {
         if (!GeNNReady) {
             gennError("You need to call initGeNN first.");
         }
         if (final) {
             gennError("Trying to add a current source to a finalized model.");
         }
+        auto targetGroup = findNeuronGroup(targetNeuronGroupName);
 
-#ifdef MPI_ENABLE
-        // Determine the host ID
-        int mpiHostID = 0;
-        MPI_Comm_rank(MPI_COMM_WORLD, &mpiHostID);
-
-        // Pick map to add group to appropriately
-        auto &groupMap = (hostID == mpiHostID) ? m_LocalCurrentSources : m_RemoteCurrentSources;
-#else
-        // If MPI is disabled always add to local neuron groups and zero host id
-        auto &groupMap = m_LocalCurrentSources;
-        hostID = 0;
-#endif
-
-        // Add neuron group to map
-        auto result = groupMap.emplace(std::piecewise_construct,
-            std::forward_as_tuple(name),
-            std::forward_as_tuple(name, model,
-                                  paramValues.getValues(), varInitialisers.getInitialisers(), hostID, deviceID));
+        // Add current source to map
+        auto result = m_CurrentSources.emplace(std::piecewise_construct,
+            std::forward_as_tuple(currentSourceName),
+            std::forward_as_tuple(currentSourceName, model,
+                                  paramValues.getValues(), varInitialisers.getInitialisers()));
 
         if(!result.second)
         {
-            gennError("Cannot add a current source with duplicate name:" + name);
-            return NULL;
+            auto source = findCurrentSource(currentSourceName);
```

Comment:
I get what you're trying to do here, but it's a bit inconsistent with the other types of thing you can add to GeNN models. For your use case it should be trivial to make the names unique in PyNN.

Reply:
It's not because of PyNN, but rather to allow injecting one current source in several populations, but I agree, in case of existing current source the API which takes all current source parameters and than ignores them looks weird.

Reply:
Especially if the new one you've added has totally different parameters, just the same name :fearful: